### ISSUE TITLE
Integration of single cell and spatial transcriptomics data

### ISSUE THREAD

**Issue by milka1g** [bowang-lab/scGPT#90]
Hello! First of all thank you for your work. It is very interesting!

I have some spatial transcriptomics (ST) datasets that are known to have lower gene expression counts. So I thought it would be a good idea to make an integration with known single cell data of that tissue and get good embeddings for downstream analysis. I have mouse brain data. For single cell data I am using [mousebrain.org data](http://mousebrain.org/adolescent/downloads.html), and for ST I have some cell-binned Stereo-seq data. (Stereo-seq has resolution lower than one cell so the common procedure is to do square bins or cell bins based on nuclei image to aggregate chip nanopores) I created an anndata file with 2 batches, one is single-cell and the other ST. I filtered both samples for the genes that are present in the scGPT vocabulary and used cells whose cell types are common to both datasets. I used the `Tutorial_Integration.ipynb` as is, replacing only the data loading part. Here are the results. 
![image](https://github.com/bowang-lab/scGPT/assets/58652409/435c3024-f8f1-42d3-89b9-027a0016ba8d)

I am wondering if tuning some of the hyperparameters would yield better results. Is it maybe difficult for the fine-tuning to work because ST expressions are lowered compared to single-cell? Is it problematic that we are mixing these two technologies and two species (mouse vs human)? Would the pretraining with mixed single-cell and ST data create a better foundation model? Would it make sense to e.g. use two ST samples and try to integrate them while using pretrained model on single-cell data? I would really appreciate any suggestions. Thanks!

Best,
Nikola

**Comment by subercui** [comment]
Hi Nikola, thank you for the interesting question. As you mentioned, it looks indeed quite challenging because of the unseen mouse data and the sptial profiling. Both were not included in the original scope of our method.

We are actually working on an new pretrained model specially for spatial transcriptomics. I hope that will help your application in the future.

For now, I would suggest: (1) How did you map the mouse genes? You can probably find the set of homologous genes between mouse and human, and then use the set to match the mouse genes to the vocabulary of pretrained scGPT. In this way, the pretrained gene embeddings can be used on these mouse genes and may help the fine-tuning. I don't have a script with me to process homologous genes, but you may search for some online resources. For example, https://useast.ensembl.org/info/website/tutorials/compara.html (2) For the same reason, I would suggest you select highly variable genes in the combined data and make sure all selected genes are these homologous genes. (3) To anwser your question, I think it is worth trying integrating multiple ST samples using the pretrained model on scRNA-seq data.

By the way what does the batch number indicate in your figure? It seems the model does a better job for batch 0 than batch 1.

**Comment by milka1g** [comment]
Hi @subercui, thank you for the insightful comments. I am really looking forward to that ST pretrained model. 
1) For the genes, I took only the intersecting genes between the scGPT vocabulary and mouse datasets. Out of ~20000 genes in the mouse datasets, ~14000 were intersecting. Since I followed the tutorial for integration where in one .h5ad file you have 2 batches of PBMC 10K that are integrated together, I marked my single-cell data as batch 0, and ST data as 1. So indeed, scGPT did better for batch 0 which is single cell data. 2) As you said, HVGs were selected in this joined dataset. I will certainly experiment more and share the results if I get some interesting ones.
